sim/druid/items.go

```diff
@@ -197,14 +197,61 @@ var ItemSetDeepEarthRegalia = core.NewItemSet(core.ItemSet{
 	Bonuses: map[int32]core.ApplyEffect{
 		// Insect Swarm increases all damage done by your Starfire, Starsurge, and Wrath spells against that target by 3%
 		2: func(agent core.Agent) {
+			const (
+				DDBC_2pcT13 int = iota
+
+				DDBC_Total
+			)
+
+			t13InsectSwarmBonus := func(_ *core.Simulation, spell *core.Spell, _ *core.AttackTable) float64 {
+				if spell.ClassSpellMask&(DruidSpellStarsurge|DruidSpellStarfire|DruidSpellWrath) > 0 {
+					return 1.03
+				}
+
+				return 1.0
+			}
+
+			druid := agent.(DruidAgent).GetDruid()
+
+			t13InsectSwarmBonusDummyAuras := druid.NewEnemyAuraArray(func(target *core.Unit) *core.Aura {
+				return target.GetOrRegisterAura(core.Aura{
+					ActionID: core.ActionID{SpellID: 105722},
+					Label:    "Item - Druid T13 Balance 2P Bonus (Insect Swarm) - " + druid.Label,
+					Duration: core.NeverExpires,
+					OnGain: func(aura *core.Aura, sim *core.Simulation) {
+						core.EnableDamageDoneByCaster(DDBC_2pcT13, DDBC_Total, druid.AttackTables[aura.Unit.UnitIndex], t13InsectSwarmBonus)
```

Comment:
Since this class only has a single `DynamicDamageDoneByCaster` that can ever be active, it will be cleaner and more performant to not mess with allocating and indexing a `DamageDoneByCasterExtraMultiplier` slice, and instead just set `DamageDoneByCasterMultiplier` directly:
```druid.AttackTables[aura.Unit.UnitIndex].DamageDoneByCasterMultiplier = t13InsectSwarmBonus```